extensions/documents/homie_last_updated_extension.md

```diff
@@ -0,0 +1,134 @@
+# Last-updated timestamp
+
+Version: **<!--VERSION-->1.0.0<!--VERSION-->**
+Date: **<!--DATE-->17. Feb 2020<!--DATE-->**
+Authors: **<!--AUTHORS-->Max Berger<!--AUTHORS-->**
+License: **<!--LICENSE-->[CCA 4.0](https://homieiot.github.io/license)<!--LICENSE-->**
+
+## Abstract
+This extension adds best-effort last-updated timestamps to properties.
+
+Homie devices will publish a last-updated timestamp along with any property value changes. The last-update timestamp can be sent by devices or gateways that do not constantly pull data from the real back-end. In this case, they can report when they last actually updated the data, so that controllers can act accordingly, and maybe ignore old data, or display this information in an UI.
+
+A common use case for this is a gateway software which publishes device data in Homie format, but actually uses external sensors to get the actual data. Gateways may re-publish the values the gather from the backend, even if these values are older. Using the last-update timestamp, a controller is then able to distinguish between actual value updates, and value updates which are just a republication of older values. 
+
+## Homie Version
+This extension supports Homie `3.0.1` and `4.x`.
+
+## Extension Identifier
+
+There are two extensions in this specification. 
+
+The ID of this first extenstion is `org.homie.ext.last-updated`.
+Therefore the **$extensions entry** is `org.homie.ext.last-updated:1.0.0:[3.0.1;4.x]`.
+
+The ID of this extension is `org.homie.ext.last-value-updated`.
+Therefore the **$extensions entry** is `org.homie.ext.last-value-updated:1.0.0:[3.0.1;4.x]`.
+
+A device may implement zero, one, or both extensions. The extension entries refer to the matching property attributes.
+
+
+## Extension Datatypes
+This extension defines no new datatypes.
+
+## Extension Attributes
+
+### Device Attributes
+
+This extension defines no device attributes.
+
+### Node Attributes
+This extension defines no direct node attributes.
+
+#### Nested Node Attributes
+
+This extension defines no nested node attributes.
+
+
+### Property Attributes
+
+This extension defines two new property attributes.
+
+#### $last-updated
+
+The **$last-updated** property attribute is **optional**.
```

Comment:
"Optional" renders the whole extension ambiguous.

Let's define the following: A Homie device may implement neither of the topics, one topic, or both topics. A Homie controller supporting the extension must support both topics.

With that the rule is simple and deterministic:
A Homie controller supporting the extension shall always evaluate `$last-value-updated` when defined, otherwise `$last-updated` and the property value.

If we agree so far, we can go one step further:
Both topics can be combined into one as the controller is able to parse which of the two formats is used and there is no case in which it actually makes sense for a Homie device to publish both formats.
Wdyt?